### PR TITLE
rar: Update to 6.10, fix livecheck, and add arch checks

### PR DESCRIPTION
**Pull request by Kreeblah**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/rar.rb [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
   name "RAR Archiver"
   desc "Archive manager for data compression and backups"
   homepage "https://www.rarlab.com/"
 
   livecheck do
-    url "https://www.rarlab.com/download.htm"
+    url "https://www.rarlab.com/rarnew.htm"
     strategy :page_match
-    regex(%r{href=.*?/rarosx-(\d+(:?\.\d+)*)\.tar\.gz}i)
+    regex(/Version\s(\d\.\d+)/i)
```
Comment (neutric) [inline Casks/rar.rb:20]: `strategy :page_match` can be deleted as it is the default except for special cases such as GitHub-URLs.

I'd stick with `download.htm` and do `regex(/>s*RAR\s+v?(\d+(:?\.\d+)+)\s+for\s+macOS/i)` as the download page probably updates closer to download availability than the changelog.

Comment (Kreeblah) [reply Casks/rar.rb]: Fair enough.  I just made those changes.

**Comment by Nothing4You** [comment]
https://www.rarlab.com/rar/rarmacos-arm-610.tar.gz results in SHA256 `64eefac221a5e5f93934360114e5e0f1f750606e0e5d7823c780887f6630c13c` rather than `db20d61544d5a569de718eac5f652a9105579db9f48ecc4ade78556a3ac24c1d` for me.

**Comment by Nothing4You** [comment]
seems they both got their archives replaced :|

**Comment by Nothing4You** [comment]
checksum update in https://github.com/Homebrew/homebrew-cask/pull/118131

**Pull request merged**